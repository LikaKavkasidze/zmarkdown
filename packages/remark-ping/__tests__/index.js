import dedent from 'dedent'
import unified from 'unified'
import reParse from 'remark-parse'
import remarkStringify from 'remark-stringify'
import remark2rehype from 'remark-rehype'
import rehypeStringify from 'rehype-stringify'

import plugin from '../src/'

const mockUsernames = [
  'I AM CLEM',
  'qux',
  'foo',
  'bar',
  'baz baz',
  'Moté',
  'Phigger Moté',
  'Digitals@m',
]

function pingUsername (username) {
  return mockUsernames.includes(username)
}
function userURL (username) {
  return `/membres/voir/${username}/`
}

const remark = text => unified()
  .use(reParse)
  .use(plugin, {pingUsername, userURL})
  .parse(text)

const toHTML = text => unified()
  .use(reParse)
  .use(plugin, {pingUsername, userURL})
  .use(remark2rehype)
  .use(rehypeStringify)
  .process(text)

const toMarkdown = text => unified()
  .use(reParse)
  .use(remarkStringify)
  .use(plugin, {pingUsername, userURL})
  .processSync(text)
  .toString()

const fixtures = [
  dedent`
    ping @Clem

    ping @**FOO BAR**

    no ping @quxjhdshqjkhfyhefezhjzjhdsjlfjlsqjdfjhsd

    ping [@**I AM CLEM**](http://example.com)

    @**baz baz**
  `,
  dedent`
    ## Test ping @**I AM CLEM**

    > > no metadata output @**I AM CLEM**

    > no metadata output @**I AM CLEM**

    ping @**I AM CLEM**

    ping _@**I AM CLEM**_

    > no metadata output @**I AM CLEM**
  `,
  dedent`
    @foo @bar

    @baz baz

    > @**baz baz**
  `,
  dedent`
    @Moté @Phigger

    @**Phigger Moté**

    @Digitals@m @**Digitals@m**
  `,
]

const pings = [
  ['I AM CLEM', 'baz baz'],
  ['I AM CLEM', 'I AM CLEM', 'I AM CLEM'],
  ['foo', 'bar'],
  ['Moté', 'Phigger Moté', 'Digitals@m'],
]

fixtures.forEach((fixture, i) => {
  describe(`fixture suite ${i}`, () => {
    test('parses', () => {
      expect(remark(fixture)).toMatchSnapshot(`f${i}`)
    })

    test('sets ping data on vfile', () => {
      return expect(
        toHTML(fixture).then(vfile => vfile.data.ping)
      ).resolves.toEqual(pings[i])
    })

    test('compiles to HTML', () => {
      return expect(
        toHTML(fixture).then(vfile => vfile.contents)
      ).resolves.toMatchSnapshot(`h${i}`)
    })

    test('compiles to Markdown', () => {
      expect(toMarkdown(fixture)).toMatchSnapshot(`m${i}`)
    })
  })
})

test('compiles to Markdown', () => {
  const toMarkdown = text => unified()
    .use(reParse)
    .use(remarkStringify)
    .use(plugin, {
      pingUsername: 12,
      userURL,
    })
    .processSync(text)
    .toString()

  expect(() => toMarkdown(dedent`
    # foo
    @**I AM CLEM**
  `)).toThrowErrorMatchingSnapshot()
})

test('do not create ping links in links', () => {
  return expect(
    toHTML(dedent`
      [foo @**I AM CLEM** bar](http://example.com)
    `).then(vfile => vfile.contents)
  ).resolves.toBe(dedent`
    <p><a href="http://example.com">foo <span class="ping ping-in-link">\
    @<span class="ping-username">I AM CLEM</span></span> bar</a></p>`)
})
