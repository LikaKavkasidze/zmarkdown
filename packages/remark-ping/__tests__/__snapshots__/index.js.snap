// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compiles to Markdown 1`] = `
"remark-ping: expected configuration to be passed: {
  pingUsername: (username) => bool,
  userURL: (username) => string
}"
`;

exports[`fixture suite 0 compiles to HTML: h0 1`] = `
"<p>ping @Clem</p>
<p>ping @<strong>FOO BAR</strong></p>
<p>no ping @quxjhdshqjkhfyhefezhjzjhdsjlfjlsqjdfjhsd</p>
<p>ping <a href=\\"http://example.com\\"><span class=\\"ping ping-in-link\\">@<span class=\\"ping-username\\">I AM CLEM</span></span></a></p>
<p><a href=\\"/membres/voir/baz baz/\\" rel=\\"nofollow\\" class=\\"ping ping-link\\">@<span class=\\"ping-username\\">baz baz</span></a></p>"
`;

exports[`fixture suite 0 compiles to Markdown: m0 1`] = `
"ping @Clem

ping @**FOO BAR**

no ping @quxjhdshqjkhfyhefezhjzjhdsjlfjlsqjdfjhsd

ping [@**I AM CLEM**](http://example.com)

@**baz baz**
"
`;

exports[`fixture suite 0 parses: f0 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "position": Position {
            "end": Object {
              "column": 11,
              "line": 1,
              "offset": 10,
            },
            "indent": Array [],
            "start": Object {
              "column": 1,
              "line": 1,
              "offset": 0,
            },
          },
          "type": "text",
          "value": "ping @Clem",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 11,
          "line": 1,
          "offset": 10,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "paragraph",
    },
    Object {
      "children": Array [
        Object {
          "position": Position {
            "end": Object {
              "column": 7,
              "line": 3,
              "offset": 18,
            },
            "indent": Array [],
            "start": Object {
              "column": 1,
              "line": 3,
              "offset": 12,
            },
          },
          "type": "text",
          "value": "ping @",
        },
        Object {
          "children": Array [
            Object {
              "position": Position {
                "end": Object {
                  "column": 16,
                  "line": 3,
                  "offset": 27,
                },
                "indent": Array [],
                "start": Object {
                  "column": 9,
                  "line": 3,
                  "offset": 20,
                },
              },
              "type": "text",
              "value": "FOO BAR",
            },
          ],
          "position": Position {
            "end": Object {
              "column": 18,
              "line": 3,
              "offset": 29,
            },
            "indent": Array [],
            "start": Object {
              "column": 7,
              "line": 3,
              "offset": 18,
            },
          },
          "type": "strong",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 18,
          "line": 3,
          "offset": 29,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 3,
          "offset": 12,
        },
      },
      "type": "paragraph",
    },
    Object {
      "children": Array [
        Object {
          "position": Position {
            "end": Object {
              "column": 50,
              "line": 5,
              "offset": 80,
            },
            "indent": Array [],
            "start": Object {
              "column": 1,
              "line": 5,
              "offset": 31,
            },
          },
          "type": "text",
          "value": "no ping @quxjhdshqjkhfyhefezhjzjhdsjlfjlsqjdfjhsd",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 50,
          "line": 5,
          "offset": 80,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 5,
          "offset": 31,
        },
      },
      "type": "paragraph",
    },
    Object {
      "children": Array [
        Object {
          "position": Position {
            "end": Object {
              "column": 6,
              "line": 7,
              "offset": 87,
            },
            "indent": Array [],
            "start": Object {
              "column": 1,
              "line": 7,
              "offset": 82,
            },
          },
          "type": "text",
          "value": "ping ",
        },
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "@",
                },
                Object {
                  "children": Array [
                    Object {
                      "type": "text",
                      "value": "I AM CLEM",
                    },
                  ],
                  "data": Object {
                    "hName": "span",
                    "hProperties": Object {
                      "class": "ping-username",
                    },
                  },
                  "type": "emphasis",
                },
              ],
              "data": Object {
                "hName": "a",
                "hProperties": Object {
                  "class": "ping ping-link",
                  "href": "/membres/voir/I AM CLEM/",
                  "rel": "nofollow",
                },
              },
              "position": Position {
                "end": Object {
                  "column": 21,
                  "line": 7,
                  "offset": 102,
                },
                "indent": Array [],
                "start": Object {
                  "column": 7,
                  "line": 7,
                  "offset": 88,
                },
              },
              "type": "ping",
              "url": "/membres/voir/I AM CLEM/",
              "username": "I AM CLEM",
            },
          ],
          "position": Position {
            "end": Object {
              "column": 42,
              "line": 7,
              "offset": 123,
            },
            "indent": Array [],
            "start": Object {
              "column": 6,
              "line": 7,
              "offset": 87,
            },
          },
          "title": null,
          "type": "link",
          "url": "http://example.com",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 42,
          "line": 7,
          "offset": 123,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 7,
          "offset": 82,
        },
      },
      "type": "paragraph",
    },
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "type": "text",
              "value": "@",
            },
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "baz baz",
                },
              ],
              "data": Object {
                "hName": "span",
                "hProperties": Object {
                  "class": "ping-username",
                },
              },
              "type": "emphasis",
            },
          ],
          "data": Object {
            "hName": "a",
            "hProperties": Object {
              "class": "ping ping-link",
              "href": "/membres/voir/baz baz/",
              "rel": "nofollow",
            },
          },
          "position": Position {
            "end": Object {
              "column": 13,
              "line": 9,
              "offset": 137,
            },
            "indent": Array [],
            "start": Object {
              "column": 1,
              "line": 9,
              "offset": 125,
            },
          },
          "type": "ping",
          "url": "/membres/voir/baz baz/",
          "username": "baz baz",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 13,
          "line": 9,
          "offset": 137,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 9,
          "offset": 125,
        },
      },
      "type": "paragraph",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 13,
      "line": 9,
      "offset": 137,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`fixture suite 1 compiles to HTML: h1 1`] = `
"<h2>Test ping <a href=\\"/membres/voir/I AM CLEM/\\" rel=\\"nofollow\\" class=\\"ping ping-link\\">@<span class=\\"ping-username\\">I AM CLEM</span></a></h2>
<blockquote>
<blockquote>
<p>no metadata output <a href=\\"/membres/voir/I AM CLEM/\\" rel=\\"nofollow\\" class=\\"ping ping-link\\">@<span class=\\"ping-username\\">I AM CLEM</span></a></p>
</blockquote>
</blockquote>
<blockquote>
<p>no metadata output <a href=\\"/membres/voir/I AM CLEM/\\" rel=\\"nofollow\\" class=\\"ping ping-link\\">@<span class=\\"ping-username\\">I AM CLEM</span></a></p>
</blockquote>
<p>ping <a href=\\"/membres/voir/I AM CLEM/\\" rel=\\"nofollow\\" class=\\"ping ping-link\\">@<span class=\\"ping-username\\">I AM CLEM</span></a></p>
<p>ping <em><a href=\\"/membres/voir/I AM CLEM/\\" rel=\\"nofollow\\" class=\\"ping ping-link\\">@<span class=\\"ping-username\\">I AM CLEM</span></a></em></p>
<blockquote>
<p>no metadata output <a href=\\"/membres/voir/I AM CLEM/\\" rel=\\"nofollow\\" class=\\"ping ping-link\\">@<span class=\\"ping-username\\">I AM CLEM</span></a></p>
</blockquote>"
`;

exports[`fixture suite 1 compiles to Markdown: m1 1`] = `
"## Test ping @**I AM CLEM**

> > no metadata output @**I AM CLEM**

> no metadata output @**I AM CLEM**

ping @**I AM CLEM**

ping _@**I AM CLEM**_

> no metadata output @**I AM CLEM**
"
`;

exports[`fixture suite 1 parses: f1 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "position": Position {
            "end": Object {
              "column": 14,
              "line": 1,
              "offset": 13,
            },
            "indent": Array [],
            "start": Object {
              "column": 4,
              "line": 1,
              "offset": 3,
            },
          },
          "type": "text",
          "value": "Test ping ",
        },
        Object {
          "children": Array [
            Object {
              "type": "text",
              "value": "@",
            },
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "I AM CLEM",
                },
              ],
              "data": Object {
                "hName": "span",
                "hProperties": Object {
                  "class": "ping-username",
                },
              },
              "type": "emphasis",
            },
          ],
          "data": Object {
            "hName": "a",
            "hProperties": Object {
              "class": "ping ping-link",
              "href": "/membres/voir/I AM CLEM/",
              "rel": "nofollow",
            },
          },
          "position": Position {
            "end": Object {
              "column": 28,
              "line": 1,
              "offset": 27,
            },
            "indent": Array [],
            "start": Object {
              "column": 14,
              "line": 1,
              "offset": 13,
            },
          },
          "type": "ping",
          "url": "/membres/voir/I AM CLEM/",
          "username": "I AM CLEM",
        },
      ],
      "depth": 2,
      "position": Position {
        "end": Object {
          "column": 28,
          "line": 1,
          "offset": 27,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "heading",
    },
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "position": Position {
                    "end": Object {
                      "column": 24,
                      "line": 3,
                      "offset": 52,
                    },
                    "indent": Array [],
                    "start": Object {
                      "column": 5,
                      "line": 3,
                      "offset": 33,
                    },
                  },
                  "type": "text",
                  "value": "no metadata output ",
                },
                Object {
                  "children": Array [
                    Object {
                      "type": "text",
                      "value": "@",
                    },
                    Object {
                      "children": Array [
                        Object {
                          "type": "text",
                          "value": "I AM CLEM",
                        },
                      ],
                      "data": Object {
                        "hName": "span",
                        "hProperties": Object {
                          "class": "ping-username",
                        },
                      },
                      "type": "emphasis",
                    },
                  ],
                  "data": Object {
                    "hName": "a",
                    "hProperties": Object {
                      "class": "ping ping-link",
                      "href": "/membres/voir/I AM CLEM/",
                      "rel": "nofollow",
                    },
                  },
                  "position": Position {
                    "end": Object {
                      "column": 38,
                      "line": 3,
                      "offset": 66,
                    },
                    "indent": Array [],
                    "start": Object {
                      "column": 24,
                      "line": 3,
                      "offset": 52,
                    },
                  },
                  "type": "ping",
                  "url": "/membres/voir/I AM CLEM/",
                  "username": "I AM CLEM",
                },
              ],
              "position": Position {
                "end": Object {
                  "column": 38,
                  "line": 3,
                  "offset": 66,
                },
                "indent": Array [],
                "start": Object {
                  "column": 5,
                  "line": 3,
                  "offset": 33,
                },
              },
              "type": "paragraph",
            },
          ],
          "position": Position {
            "end": Object {
              "column": 38,
              "line": 3,
              "offset": 66,
            },
            "indent": Array [],
            "start": Object {
              "column": 3,
              "line": 3,
              "offset": 31,
            },
          },
          "type": "blockquote",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 38,
          "line": 3,
          "offset": 66,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 3,
          "offset": 29,
        },
      },
      "type": "blockquote",
    },
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "position": Position {
                "end": Object {
                  "column": 22,
                  "line": 5,
                  "offset": 89,
                },
                "indent": Array [],
                "start": Object {
                  "column": 3,
                  "line": 5,
                  "offset": 70,
                },
              },
              "type": "text",
              "value": "no metadata output ",
            },
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "@",
                },
                Object {
                  "children": Array [
                    Object {
                      "type": "text",
                      "value": "I AM CLEM",
                    },
                  ],
                  "data": Object {
                    "hName": "span",
                    "hProperties": Object {
                      "class": "ping-username",
                    },
                  },
                  "type": "emphasis",
                },
              ],
              "data": Object {
                "hName": "a",
                "hProperties": Object {
                  "class": "ping ping-link",
                  "href": "/membres/voir/I AM CLEM/",
                  "rel": "nofollow",
                },
              },
              "position": Position {
                "end": Object {
                  "column": 36,
                  "line": 5,
                  "offset": 103,
                },
                "indent": Array [],
                "start": Object {
                  "column": 22,
                  "line": 5,
                  "offset": 89,
                },
              },
              "type": "ping",
              "url": "/membres/voir/I AM CLEM/",
              "username": "I AM CLEM",
            },
          ],
          "position": Position {
            "end": Object {
              "column": 36,
              "line": 5,
              "offset": 103,
            },
            "indent": Array [],
            "start": Object {
              "column": 3,
              "line": 5,
              "offset": 70,
            },
          },
          "type": "paragraph",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 36,
          "line": 5,
          "offset": 103,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 5,
          "offset": 68,
        },
      },
      "type": "blockquote",
    },
    Object {
      "children": Array [
        Object {
          "position": Position {
            "end": Object {
              "column": 6,
              "line": 7,
              "offset": 110,
            },
            "indent": Array [],
            "start": Object {
              "column": 1,
              "line": 7,
              "offset": 105,
            },
          },
          "type": "text",
          "value": "ping ",
        },
        Object {
          "children": Array [
            Object {
              "type": "text",
              "value": "@",
            },
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "I AM CLEM",
                },
              ],
              "data": Object {
                "hName": "span",
                "hProperties": Object {
                  "class": "ping-username",
                },
              },
              "type": "emphasis",
            },
          ],
          "data": Object {
            "hName": "a",
            "hProperties": Object {
              "class": "ping ping-link",
              "href": "/membres/voir/I AM CLEM/",
              "rel": "nofollow",
            },
          },
          "position": Position {
            "end": Object {
              "column": 20,
              "line": 7,
              "offset": 124,
            },
            "indent": Array [],
            "start": Object {
              "column": 6,
              "line": 7,
              "offset": 110,
            },
          },
          "type": "ping",
          "url": "/membres/voir/I AM CLEM/",
          "username": "I AM CLEM",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 20,
          "line": 7,
          "offset": 124,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 7,
          "offset": 105,
        },
      },
      "type": "paragraph",
    },
    Object {
      "children": Array [
        Object {
          "position": Position {
            "end": Object {
              "column": 6,
              "line": 9,
              "offset": 131,
            },
            "indent": Array [],
            "start": Object {
              "column": 1,
              "line": 9,
              "offset": 126,
            },
          },
          "type": "text",
          "value": "ping ",
        },
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "@",
                },
                Object {
                  "children": Array [
                    Object {
                      "type": "text",
                      "value": "I AM CLEM",
                    },
                  ],
                  "data": Object {
                    "hName": "span",
                    "hProperties": Object {
                      "class": "ping-username",
                    },
                  },
                  "type": "emphasis",
                },
              ],
              "data": Object {
                "hName": "a",
                "hProperties": Object {
                  "class": "ping ping-link",
                  "href": "/membres/voir/I AM CLEM/",
                  "rel": "nofollow",
                },
              },
              "position": Position {
                "end": Object {
                  "column": 21,
                  "line": 9,
                  "offset": 146,
                },
                "indent": Array [],
                "start": Object {
                  "column": 7,
                  "line": 9,
                  "offset": 132,
                },
              },
              "type": "ping",
              "url": "/membres/voir/I AM CLEM/",
              "username": "I AM CLEM",
            },
          ],
          "position": Position {
            "end": Object {
              "column": 22,
              "line": 9,
              "offset": 147,
            },
            "indent": Array [],
            "start": Object {
              "column": 6,
              "line": 9,
              "offset": 131,
            },
          },
          "type": "emphasis",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 22,
          "line": 9,
          "offset": 147,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 9,
          "offset": 126,
        },
      },
      "type": "paragraph",
    },
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "position": Position {
                "end": Object {
                  "column": 22,
                  "line": 11,
                  "offset": 170,
                },
                "indent": Array [],
                "start": Object {
                  "column": 3,
                  "line": 11,
                  "offset": 151,
                },
              },
              "type": "text",
              "value": "no metadata output ",
            },
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "@",
                },
                Object {
                  "children": Array [
                    Object {
                      "type": "text",
                      "value": "I AM CLEM",
                    },
                  ],
                  "data": Object {
                    "hName": "span",
                    "hProperties": Object {
                      "class": "ping-username",
                    },
                  },
                  "type": "emphasis",
                },
              ],
              "data": Object {
                "hName": "a",
                "hProperties": Object {
                  "class": "ping ping-link",
                  "href": "/membres/voir/I AM CLEM/",
                  "rel": "nofollow",
                },
              },
              "position": Position {
                "end": Object {
                  "column": 36,
                  "line": 11,
                  "offset": 184,
                },
                "indent": Array [],
                "start": Object {
                  "column": 22,
                  "line": 11,
                  "offset": 170,
                },
              },
              "type": "ping",
              "url": "/membres/voir/I AM CLEM/",
              "username": "I AM CLEM",
            },
          ],
          "position": Position {
            "end": Object {
              "column": 36,
              "line": 11,
              "offset": 184,
            },
            "indent": Array [],
            "start": Object {
              "column": 3,
              "line": 11,
              "offset": 151,
            },
          },
          "type": "paragraph",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 36,
          "line": 11,
          "offset": 184,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 11,
          "offset": 149,
        },
      },
      "type": "blockquote",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 36,
      "line": 11,
      "offset": 184,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`fixture suite 2 compiles to HTML: h2 1`] = `
"<p><a href=\\"/membres/voir/foo/\\" rel=\\"nofollow\\" class=\\"ping ping-link\\">@<span class=\\"ping-username\\">foo</span></a> <a href=\\"/membres/voir/bar/\\" rel=\\"nofollow\\" class=\\"ping ping-link\\">@<span class=\\"ping-username\\">bar</span></a></p>
<p>@baz baz</p>
<blockquote>
<p><a href=\\"/membres/voir/baz baz/\\" rel=\\"nofollow\\" class=\\"ping ping-link\\">@<span class=\\"ping-username\\">baz baz</span></a></p>
</blockquote>"
`;

exports[`fixture suite 2 compiles to Markdown: m2 1`] = `
"@foo @bar

@baz baz

> @**baz baz**
"
`;

exports[`fixture suite 2 parses: f2 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "type": "text",
              "value": "@",
            },
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "foo",
                },
              ],
              "data": Object {
                "hName": "span",
                "hProperties": Object {
                  "class": "ping-username",
                },
              },
              "type": "emphasis",
            },
          ],
          "data": Object {
            "hName": "a",
            "hProperties": Object {
              "class": "ping ping-link",
              "href": "/membres/voir/foo/",
              "rel": "nofollow",
            },
          },
          "position": Position {
            "end": Object {
              "column": 5,
              "line": 1,
              "offset": 4,
            },
            "indent": Array [],
            "start": Object {
              "column": 1,
              "line": 1,
              "offset": 0,
            },
          },
          "type": "ping",
          "url": "/membres/voir/foo/",
          "username": "foo",
        },
        Object {
          "position": Position {
            "end": Object {
              "column": 6,
              "line": 1,
              "offset": 5,
            },
            "indent": Array [],
            "start": Object {
              "column": 5,
              "line": 1,
              "offset": 4,
            },
          },
          "type": "text",
          "value": " ",
        },
        Object {
          "children": Array [
            Object {
              "type": "text",
              "value": "@",
            },
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "bar",
                },
              ],
              "data": Object {
                "hName": "span",
                "hProperties": Object {
                  "class": "ping-username",
                },
              },
              "type": "emphasis",
            },
          ],
          "data": Object {
            "hName": "a",
            "hProperties": Object {
              "class": "ping ping-link",
              "href": "/membres/voir/bar/",
              "rel": "nofollow",
            },
          },
          "position": Position {
            "end": Object {
              "column": 10,
              "line": 1,
              "offset": 9,
            },
            "indent": Array [],
            "start": Object {
              "column": 6,
              "line": 1,
              "offset": 5,
            },
          },
          "type": "ping",
          "url": "/membres/voir/bar/",
          "username": "bar",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 10,
          "line": 1,
          "offset": 9,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "paragraph",
    },
    Object {
      "children": Array [
        Object {
          "position": Position {
            "end": Object {
              "column": 9,
              "line": 3,
              "offset": 19,
            },
            "indent": Array [],
            "start": Object {
              "column": 1,
              "line": 3,
              "offset": 11,
            },
          },
          "type": "text",
          "value": "@baz baz",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 9,
          "line": 3,
          "offset": 19,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 3,
          "offset": 11,
        },
      },
      "type": "paragraph",
    },
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "@",
                },
                Object {
                  "children": Array [
                    Object {
                      "type": "text",
                      "value": "baz baz",
                    },
                  ],
                  "data": Object {
                    "hName": "span",
                    "hProperties": Object {
                      "class": "ping-username",
                    },
                  },
                  "type": "emphasis",
                },
              ],
              "data": Object {
                "hName": "a",
                "hProperties": Object {
                  "class": "ping ping-link",
                  "href": "/membres/voir/baz baz/",
                  "rel": "nofollow",
                },
              },
              "position": Position {
                "end": Object {
                  "column": 15,
                  "line": 5,
                  "offset": 35,
                },
                "indent": Array [],
                "start": Object {
                  "column": 3,
                  "line": 5,
                  "offset": 23,
                },
              },
              "type": "ping",
              "url": "/membres/voir/baz baz/",
              "username": "baz baz",
            },
          ],
          "position": Position {
            "end": Object {
              "column": 15,
              "line": 5,
              "offset": 35,
            },
            "indent": Array [],
            "start": Object {
              "column": 3,
              "line": 5,
              "offset": 23,
            },
          },
          "type": "paragraph",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 15,
          "line": 5,
          "offset": 35,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 5,
          "offset": 21,
        },
      },
      "type": "blockquote",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 15,
      "line": 5,
      "offset": 35,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`fixture suite 3 compiles to HTML: h3 1`] = `
"<p><a href=\\"/membres/voir/Moté/\\" rel=\\"nofollow\\" class=\\"ping ping-link\\">@<span class=\\"ping-username\\">Moté</span></a> @Phigger</p>
<p><a href=\\"/membres/voir/Phigger Moté/\\" rel=\\"nofollow\\" class=\\"ping ping-link\\">@<span class=\\"ping-username\\">Phigger Moté</span></a></p>
<p>@Digitals@m <a href=\\"/membres/voir/Digitals@m/\\" rel=\\"nofollow\\" class=\\"ping ping-link\\">@<span class=\\"ping-username\\">Digitals@m</span></a></p>"
`;

exports[`fixture suite 3 compiles to Markdown: m3 1`] = `
"@Moté @Phigger

@**Phigger Moté**

@Digitals@m @**Digitals@m**
"
`;

exports[`fixture suite 3 parses: f3 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "type": "text",
              "value": "@",
            },
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "Moté",
                },
              ],
              "data": Object {
                "hName": "span",
                "hProperties": Object {
                  "class": "ping-username",
                },
              },
              "type": "emphasis",
            },
          ],
          "data": Object {
            "hName": "a",
            "hProperties": Object {
              "class": "ping ping-link",
              "href": "/membres/voir/Moté/",
              "rel": "nofollow",
            },
          },
          "position": Position {
            "end": Object {
              "column": 6,
              "line": 1,
              "offset": 5,
            },
            "indent": Array [],
            "start": Object {
              "column": 1,
              "line": 1,
              "offset": 0,
            },
          },
          "type": "ping",
          "url": "/membres/voir/Moté/",
          "username": "Moté",
        },
        Object {
          "position": Position {
            "end": Object {
              "column": 15,
              "line": 1,
              "offset": 14,
            },
            "indent": Array [],
            "start": Object {
              "column": 6,
              "line": 1,
              "offset": 5,
            },
          },
          "type": "text",
          "value": " @Phigger",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 15,
          "line": 1,
          "offset": 14,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "paragraph",
    },
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "type": "text",
              "value": "@",
            },
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "Phigger Moté",
                },
              ],
              "data": Object {
                "hName": "span",
                "hProperties": Object {
                  "class": "ping-username",
                },
              },
              "type": "emphasis",
            },
          ],
          "data": Object {
            "hName": "a",
            "hProperties": Object {
              "class": "ping ping-link",
              "href": "/membres/voir/Phigger Moté/",
              "rel": "nofollow",
            },
          },
          "position": Position {
            "end": Object {
              "column": 18,
              "line": 3,
              "offset": 33,
            },
            "indent": Array [],
            "start": Object {
              "column": 1,
              "line": 3,
              "offset": 16,
            },
          },
          "type": "ping",
          "url": "/membres/voir/Phigger Moté/",
          "username": "Phigger Moté",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 18,
          "line": 3,
          "offset": 33,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 3,
          "offset": 16,
        },
      },
      "type": "paragraph",
    },
    Object {
      "children": Array [
        Object {
          "position": Position {
            "end": Object {
              "column": 13,
              "line": 5,
              "offset": 47,
            },
            "indent": Array [],
            "start": Object {
              "column": 1,
              "line": 5,
              "offset": 35,
            },
          },
          "type": "text",
          "value": "@Digitals@m ",
        },
        Object {
          "children": Array [
            Object {
              "type": "text",
              "value": "@",
            },
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "Digitals@m",
                },
              ],
              "data": Object {
                "hName": "span",
                "hProperties": Object {
                  "class": "ping-username",
                },
              },
              "type": "emphasis",
            },
          ],
          "data": Object {
            "hName": "a",
            "hProperties": Object {
              "class": "ping ping-link",
              "href": "/membres/voir/Digitals@m/",
              "rel": "nofollow",
            },
          },
          "position": Position {
            "end": Object {
              "column": 28,
              "line": 5,
              "offset": 62,
            },
            "indent": Array [],
            "start": Object {
              "column": 13,
              "line": 5,
              "offset": 47,
            },
          },
          "type": "ping",
          "url": "/membres/voir/Digitals@m/",
          "username": "Digitals@m",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 28,
          "line": 5,
          "offset": 62,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 5,
          "offset": 35,
        },
      },
      "type": "paragraph",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 28,
      "line": 5,
      "offset": 62,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;
